editors/code/rust.tmGrammar.json

```diff
@@ -1,686 +1,1048 @@
-			"name": "storage.modifier.const.rust",
-			"match": "\\bconst\\b"
-		},
-		"pub": {
-			"comment": "Visibility modifier",
-			"name": "storage.modifier.visibility.rust",
-			"match": "\\bpub\\b"
-		},
-		"unsafe": {
-			"comment": "Unsafe code keyword",
-			"name": "keyword.other.unsafe.rust",
-			"match": "\\bunsafe\\b"
-		},
-		"where": {
-			"comment": "Generic where clause",
-			"name": "keyword.other.where.rust",
-			"match": "\\bwhere\\b"
-		},
-		"lifetime": {
-			"comment": "Named lifetime",
-			"name": "storage.modifier.lifetime.rust",
-			"match": "'([a-zA-Z_][a-zA-Z0-9_]*)\\b"
-		},
-		"ref_lifetime": {
-			"comment": "Reference with named lifetime",
-			"match": "(&)('[a-zA-Z_][a-zA-Z0-9_]*)\\b",
-			"captures": {
-				"1": {
-					"name": "keyword.operator.sigil.rust"
-				},
-				"2": {
-					"name": "storage.modifier.lifetime.rust"
-				}
-			}
-		},
-		"core_types": {
-			"comment": "Built-in/core type",
-			"name": "support.type.primitive.rust",
-			"match": "\\b(bool|char|usize|isize|u8|u16|u32|u64|u128|i8|i16|i32|i64|i128|f32|f64|str|Self)\\b"
-		},
-		"core_vars": {
-			"comment": "Core type variant",
-			"name": "support.constant.core.rust",
-			"match": "\\b(Some|None|Ok|Err)\\b"
-		},
-		"core_marker": {
-			"comment": "Core trait (marker)",
-			"name": "entity.name.type.marker.rust",
-			"match": "\\b(Copy|Send|Sized|Sync)\\b"
-		},
-		"core_traits": {
-			"comment": "Core trait",
-			"name": "entity.name.type.core.rust",
-			"match": "\\b(Drop|Fn|FnMut|FnOnce|Clone|PartialEq|PartialOrd|Eq|Ord|AsRef|AsMut|Into|From|Default|Iterator|Extend|IntoIterator|DoubleEndedIterator|ExactSizeIterator)\\b"
-		},
-		"std_types": {
-			"comment": "Standard library type",
-			"name": "entity.name.type.class.std.rust",
-			"match": "\\b(Box|String|Vec|Path|PathBuf|Option|Result)\\b"
-		},
-		"std_traits": {
-			"comment": "Standard library trait",
-			"name": "entity.name.type.std.rust",
-			"match": "\\b(ToOwned|ToString)\\b"
-		},
-		"type": {
-			"comment": "A type",
-			"name": "entity.name.type.rust",
-			"match": "\\b([A-Za-z][_A-Za-z0-9]*|_[_A-Za-z0-9]+)\\b"
-		},
-		"type_params": {
-			"comment": "Type parameters",
-			"name": "meta.type_params.rust",
-			"begin": "<(?![=<])",
-			"end": "(?<![-])>",
-			"patterns": [
-				{
-					"include": "#block_comment"
-				},
-				{
-					"include": "#line_comment"
-				},
-				{
-					"include": "#sigils"
-				},
-				{
-					"include": "#mut"
-				},
-				{
-					"include": "#dyn"
-				},
-				{
-					"include": "#impl"
-				},
-				{
-					"include": "#lifetime"
-				},
-				{
-					"include": "#core_types"
-				},
-				{
-					"include": "#core_marker"
-				},
-				{
-					"include": "#core_traits"
-				},
-				{
-					"include": "#std_types"
-				},
-				{
-					"include": "#std_traits"
-				},
-				{
-					"include": "#type_params"
-				}
-			]
-		}
-	}
+  "$schema": "https://raw.githubusercontent.com/martinring/tmlanguage/master/tmlanguage.json",
```

Comment:
We use 4 space indentation everywhere.

Reply:
Fixed, thanks.

Reply:
This could be avoided by having an [.editorconfig](https://editorconfig.org/), wdyt?

Reply:
I don't think it matters. Since this grammar is maintained externally by @dustypomerleau, we could bundle it as-is instead of reformatting the JSON.

Reply:
I'm flexible on this. I am maintaining the grammar [as YAML](https://github.com/dustypomerleau/rust-syntax/blob/master/syntaxes/rust.tmLanguage.yaml) and generating the JSON.